src/Analyzer/QueryTreePassManager.h

```diff
@@ -25,15 +25,15 @@ class QueryTreePassManager : public WithContext
     void addPass(QueryTreePassPtr pass);
 
     /// Run query tree passes on query tree
-    void run(QueryTreeNodePtr & query_tree_node);
+    void run(QueryTreeNodePtr query_tree_node);
```

Comment:
@rschu1ze without the reference to the shared_ptr in the Manager, I cannot easily replace the top-level QueryNode with my wrapping QueryNode that incl the ORDER BY rand(). What do you think?